### ISSUE TITLE
Assertion in Debug mode in geojson.input

### ISSUE THREAD

**Issue by springmeyer** [mapnik/mapnik#3709]
It appears that the geojson.input plugin may trigger an assertion in boost::rtree. This may be harmless or may be a problem. Perhaps we need to protect from sending invalid input to boost::rtree. /cc @artemp 

Seen with the node-mapnik tests when build against a mapnik core (tested with v3.0.x branch) build in debug mode. (NOTE: my local build is also built with `-fsantize-address` but I presume this is replicable without that and just with `-DDEBUG`):

```
    ✓ test empty geojson datasource due to invalid json string
Assertion failed: ((detail::is_valid(indexable))&&("Indexable is invalid")), function apply, file mason_packages/.link/include/boost/geometry/index/detail/rtree/pack_create.hpp, line 177.
Abort trap: 6
```

The exact failing test is https://github.com/mapnik/node-mapnik/blob/49799c1f4a915f26b776667e09339183ef2e882e/test/datasource.test.js#L215-L217

Backtrace is:

```
Application Specific Information:
Assertion failed: ((detail::is_valid(indexable))&&("Indexable is invalid")), function apply, file mason_packages/.link/include/boost/geometry/index/detail/rtree/pack_create.hpp, line 177.
 

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff9530fd42 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff953fd457 pthread_kill + 90
2   libsystem_c.dylib             	0x00007fff95275420 abort + 129
3   libsystem_c.dylib             	0x00007fff9523c893 __assert_rtn + 320
4   geojson.input                 	0x00000001133f383d boost::variant<boost::geometry::index::detail::rtree::variant_leaf<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::allocators<std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::node_variant_static_tag>, boost::geometry::index::detail::rtree::node_variant_static_tag>, boost::geometry::index::detail::rtree::variant_internal_node<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::allocators<std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::node_variant_static_tag>, boost::geometry::index::detail::rtree::node_variant_static_tag> >* boost::geometry::index::detail::rtree::pack<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, boost::geometry::index::detail::rtree::options<geojson_linear<16ul, 4ul>, boost::geometry::index::detail::rtree::insert_default_tag, boost::geometry::index::detail::rtree::choose_by_content_diff_tag, boost::geometry::index::detail::rtree::split_default_tag, boost::geometry::index::detail::rtree::linear_tag, boost::geometry::index::detail::rtree::node_variant_static_tag>, boost::geometry::index::detail::translator<boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::allocators<std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::node_variant_static_tag> >::apply<std::__1::__wrap_iter<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > const*> >(std::__1::__wrap_iter<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > const*>, std::__1::__wrap_iter<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > const*>, unsigned long&, unsigned long&, geojson_linear<16ul, 4ul> const&, boost::geometry::index::detail::translator<boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > const&, boost::geometry::index::detail::rtree::allocators<std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::model::box<boost::geometry::model::point<double, 2ul, boost::geometry::cs::cartesian> >, boost::geometry::index::detail::rtree::node_variant_static_tag>&) + 3357 (pack_create.hpp:177)
5   geojson.input                 	0x00000001133f28fa boost::geometry::index::rtree<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >::rtree<std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > >(std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > const&, geojson_linear<16ul, 4ul> const&, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&) + 922 (rtree.hpp:332)
6   geojson.input                 	0x00000001133f254d boost::geometry::index::rtree<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >::rtree<std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > >(std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > const&, geojson_linear<16ul, 4ul> const&, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > const&) + 61 (rtree.hpp:336)
7   geojson.input                 	0x00000001133e291d std::__1::unique_ptr<boost::geometry::index::rtree<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >, std::__1::default_delete<boost::geometry::index::rtree<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > > > > std::make_unique<boost::geometry::index::rtree<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, geojson_linear<16ul, 4ul>, boost::geometry::index::indexable<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, boost::geometry::index::equal_to<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >, std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >&>(std::__1::vector<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> >, std::__1::allocator<std::__1::pair<mapnik::box2d<double>, std::__1::pair<unsigned long, unsigned long> > > >&&&) + 509 (make_unique.hpp:36)
8   geojson.input                 	0x000000011338aa69 void geojson_datasource::initialise_index<char const*>(char const*, char const*) + 12505 (geojson_datasource.cpp:304)
9   geojson.input                 	0x000000011337f0a7 geojson_datasource::geojson_datasource(mapnik::parameters const&) + 25127 (geojson_datasource.cpp:179)
10  geojson.input                 	0x0000000113378dcd geojson_datasource::geojson_datasource(mapnik::parameters const&) + 29 (geojson_datasource.cpp:182)
11  geojson.input                 	0x0000000113378d82 create + 50 (geojson_datasource.cpp:70)
12  libmapnik.dylib               	0x000000010ac361b1 mapnik::datasource_cache::create(mapnik::parameters const&) + 21617 (datasource_cache.cpp:126)
13  mapnik.node                   	0x000000010889c165 Datasource::New(Nan::FunctionCallbackInfo<v8::Value> const&) + 7813 (mapnik_datasource.cpp:112)
14  mapnik.node                   	0x00000001081d717f Nan::imp::FunctionCallbackWrapper(v8::FunctionCallbackInfo<v8::Value> const&) + 783 (nan_callbacks_12_inl.h:175)
15  node                          	0x00000001001780bf v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) + 159
16  node                          	0x00000001001a36bc v8::internal::Builtin_HandleApiCallConstruct(int, v8::internal::Object**, v8::internal::Isolate*) + 732
```

**Comment by artemp** [comment]
confirmed and fixed in a408b0732eee32cbc9e508b0593e83d7b075128d 
(this bug doesn't apply to master branch)

**Comment by springmeyer** [comment]
Great, thanks for closing out.
